testnet/stacks-node/src/tests/nakamoto_integrations.rs

```diff
@@ -11237,3 +11237,139 @@ fn reload_miner_config() {
+    btc_regtest_controller.bootstrap_chain(201);
+
+    let conf_path =
+        std::env::temp_dir().join(format!("miner-config-test-{}.toml", rand::random::<u64>()));
+    conf.config_path = Some(conf_path.clone().to_str().unwrap().to_string());
+
+    // Make a minimum-viable config file
+    let update_config = |burn_fee_cap: u64, sats_vbyte: u64| {
+        use std::io::Write;
+
+        let new_config = format!(
+            r#"
+            [burnchain]
+            burn_fee_cap = {}
+            satoshis_per_byte = {}
+            "#,
+            burn_fee_cap, sats_vbyte,
+        );
+        // Write to a file
+        let mut file = File::create(&conf_path).unwrap();
+        file.write_all(new_config.as_bytes()).unwrap();
+    };
+
+    let mut run_loop = boot_nakamoto::BootRunLoop::new(conf.clone()).unwrap();
+    let run_loop_stopper = run_loop.get_termination_switch();
+    let counters = run_loop.counters();
+    let Counters {
+        blocks_processed,
+        naka_submitted_commits: commits_submitted,
+        ..
+    } = run_loop.counters();
+
+    let coord_channel = run_loop.coordinator_channels();
+
+    let run_loop_thread = thread::spawn(move || run_loop.start(None, 0));
+    let mut signers = TestSigners::new(vec![signer_sk]);
+    wait_for_runloop(&blocks_processed);
+    boot_to_epoch_3(
+        &conf,
+        &blocks_processed,
+        &[stacker_sk],
+        &[signer_sk],
+        &mut Some(&mut signers),
+        &mut btc_regtest_controller,
+    );
+
+    info!("------------------------- Reached Epoch 3.0 -------------------------");
+
+    blind_signer(&conf, &signers, &counters);
+
+    wait_for_first_naka_block_commit(60, &commits_submitted);
+
+    next_block_and_mine_commit(&mut btc_regtest_controller, 60, &conf, &counters).unwrap();
+
+    let burnchain = conf.get_burnchain();
+    let sortdb = burnchain.open_sortition_db(true).unwrap();
+
+    let tip = SortitionDB::get_canonical_burn_chain_tip(sortdb.conn()).unwrap();
+    let stacks_height = tip.stacks_block_height;
+
+    let mut last_log = Instant::now();
+    last_log -= Duration::from_secs(5);
+    wait_for(30, || {
+        let last_commit = &counters.naka_submitted_commit_last_stacks_tip.get();
+        if last_log.elapsed() >= Duration::from_secs(5) {
+            info!(
+                "---- last_commit: {:?} stacks_height: {:?} ---- ",
+                last_commit, stacks_height
+            );
+            last_log = Instant::now();
+        }
+        Ok(*last_commit >= stacks_height)
+    })
+    .expect("Failed to wait for last commit");
+
+    let commits_before = counters.naka_submitted_commits.get();
+
+    info!("---- Updating config ----");
+
+    update_config(155000, 57);
+
+    wait_for(30, || {
+        let commit_count = &counters.naka_submitted_commits.get();
+        Ok(*commit_count > commits_before)
+    })
+    .expect("Expected new commit after config change");
```

Comment:
Could we check here to see if the latest commit uses the new values? I'm not sure what the easiest way to do that is going to be... maybe mine the next bitcoin block and then we can check for the pox amount in the DB. We may need to query the bitcoin node to get the fee.